src/libAtomVM/nifs.h

```diff
@@ -46,6 +46,12 @@ extern "C" {
     ctx->x[1] = (error_type_atom);   \
     return term_invalid_term();
 
+#define REQUIRE_ATOM(name, lenstr, str, glb) \
+    term name = globalcontext_make_atom(glb, lenstr str); \
+    if (UNLIKELY(term_is_invalid_term(name))) { \
+        RAISE_ERROR(OUT_OF_MEMORY_ATOM); \
```

Comment:
This is perfect for nifs, but if this is used in a port is raising the error what we really want? Maybe we need a separate macro for use in ports that would just return the out_of_memory atom? Ideally it would be {error, out_of_memory} (for ports), but in this case it would be unlikely there was enough ram to construct the error tuple.